From aa6e3c52d398e4e43296b628c6339af8994c6674 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Wed, 4 Apr 2012 20:36:03 +0900
Subject: [PATCH 03/39] base_filter: set timestamps if not set

Timestamps must be set in order to detect the frame boundary
according to the specification of REL OMX.
It isn't usually necessary because timestamps are set by demuxer.
---
 omx/gstomx_base_filter.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/omx/gstomx_base_filter.c b/omx/gstomx_base_filter.c
index 8b9ffd5..fa16c7e 100644
--- a/omx/gstomx_base_filter.c
+++ b/omx/gstomx_base_filter.c
@@ -649,6 +649,12 @@ pad_chain (GstPad * pad, GstBuffer * buf)
         else
           omx_buffer->nFlags = OMX_BUFFERFLAG_ENDOFFRAME;
 
+        if ((GST_BUFFER_TIMESTAMP (buf) == 0) ||
+            (GST_BUFFER_TIMESTAMP (buf) == GST_CLOCK_TIME_NONE)) {
+          static guint64 cnt = 0;
+          omx_buffer->nTimeStamp = cnt++;
+        }
+
         GST_LOG_OBJECT (self, "release_buffer");
                 /** @todo untaint buffer */
         g_omx_port_release_buffer (in_port, omx_buffer);
-- 
1.7.9.5

